amaranth/hdl/ast.py

```diff
@@ -1382,6 +1382,8 @@ def __init__(self, iterable=()):
         self._mutable  = True
 
     def __getitem__(self, index):
+        if isinstance(index, ValueCastable):
```

Comment:
You don't need this check.

Reply:
If we remove this check, we change the semantics of the method. currently if we give the method an int, we will not get an ArrayProxy. With this change, we would.

Reply:
Oh, you are right.